library/Fission/Web/Heroku.hs

```diff
@@ -110,13 +104,12 @@ provision Request {uuid, region} = do
 type DeprovisionAPI = Capture "addon_id" UUID
                    :> DeleteNoContent '[Heroku.MIME.VendorJSONv3] NoContent
 
-deprovision
-  :: ( MonadSelda   (RIO cfg)
-     , MonadMask    (RIO cfg)
-     , HasLogFunc        cfg
-     , Has HTTP.Manager  cfg
-     , Has IPFS.URL      cfg
-     )
+deprovision ::
+  ( MonadSelda      (RIO cfg)
+  , MonadMask       (RIO cfg)
+  , MonadRemoteIPFS (RIO cfg)
+  , HasLogFunc           cfg
+  )
```

Comment:
Awwwwwwwwww yeeeeeeees. We'd need to implement [one more class](https://www.stackage.org/haddock/lts-14.16/monad-logger-0.3.30/Control-Monad-Logger.html#t:MonadLogger), but check this out:

```haskell
instance HasLogFunc cfg => MonadLogger (RIO cfg) where
  monadLoggerLog _loc _src level msg =
    logger <- Config.get
    logger txt level -- or something

deprovision ::
  ( MonadSelda      m
  , MonadMask       m
  , MonadRemoteIPFS m
  , MonadLogger     m
  )
  => ServerT DeprovisionAPI m
deprovision uuid' = do
  let err = Web.Err.ensureMaybe err410 -- HTTP 410 is specified by the Heroku AddOn docs
  let err = Web.Err.ensureMaybe err410

Reply:
i.e. no more tracking `cfg` all over the place. Constraints == capabilities.

Reply:
awwww yiiisss